src/hyperlink/_url.py

```diff
@@ -219,8 +243,14 @@ def _encode_schemeless_path_part(text, maximal=True):
                      if t in _SCHEMELESS_PATH_DELIMS else t for t in text])
 
 
-def _encode_path_parts(text_parts, rooted=False, has_scheme=True,
-                       has_authority=True, joined=True, maximal=True):
+def _encode_path_parts(
+    text_parts,          # type: Sequence[Text]
+    rooted=False,        # type: bool
+    has_scheme=True,     # type: bool
+    has_authority=True,  # type: bool
+    maximal=True,        # type: bool
+):
+    # type: (...) -> Sequence[Text]
```

Comment:
Note that I removed the `joined` argument here.
Keeping it would mean that we're return either a sequence of strings or a string, which would require casting elsewhere.  It's easy enough for the caller to join the parts if needed, which IIRC I had to only fix in one place.